baseplate/server/healthcheck.py

```diff
@@ -46,6 +60,7 @@ def check_http_service(endpoint):
 
 CHECKERS = {
     "thrift": check_thrift_service,
+    "thrift-framed": check_framed_thrift_service,
```

Comment:
I'm kinda torn here. I see the benefit in this being here for your server, but a baseplate server is never going to use a different protocol like this and so it feels a little confusing/cluttery. Maybe just having the ability to use the other factory (the rest of this patch) is sufficient and we can use that code in a custom healthcheck on the puppet side for your stuff?